chips/sam4l/src/dma.rs

```diff
@@ -144,12 +153,18 @@ impl DMAChannel {
         DMAChannel {
             registers: (DMA_BASE_ADDR + (channel as usize) * DMA_CHANNEL_SIZE) as *mut DMARegisters,
             nvic: nvic,
-            client: None,
+            client: Cell::new(None),
+            width: Cell::new(DMAWidth::Width8Bit),
             enabled: Cell::new(false),
             buffer: TakeCell::empty(),
         }
     }
 
+    pub fn initialize(&self, client: &'static mut DMAClient, width: DMAWidth) {
```

Comment:
I was treating it as an initialization-time only operation for now. Similarly, when we set up capsule/driver clients in platform `main.rs` files we expect to only do so once as well and have no way to retrieve them.

I can imagine a future design where, if we run out of channels, we connect them with clients and widths at runtime, possibly changing them. I would argue for holding of on a more advanced interface (like returning clients) until that need occurs though.